puppetca/resource_puppetca_certificate.go

```diff
@@ -55,14 +75,46 @@ func resourcePuppetCACertificateCreate(d *schema.ResourceData, meta interface{})
 	return resourcePuppetCACertificateRead(d, meta)
 }
 
-func findCert(client puppetca.Client, name string) resource.StateRefreshFunc {
+func findCert(client puppetca.Client, name, env string) (string, string, error) {
+	cert, err := client.GetCertByName(name, env)
+
+	if err != nil || cert == "" {
+		if strings.Contains(err.Error(), "404 Not Found") {
```

Comment:
Does the PuppetCA always return a 200? Ideally, the code should be checked in the HTTP error, not in the body.

Reply:
Puppet returns 404 which is passed from go-puppetca and this bubble up to terraform. If we return the 404 (which is expected in this case) terraform stops and claims that the provider has a bug.
I don't know how to check the HTTP error :D but I will have a look at this.

Reply:
As per https://github.com/camptocamp/go-puppetca/blob/master/puppetca/puppetca.go#L184 we do not have the status code of the response. Neither the response nor the err are returned
```golang
	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusNoContent {
		return "", fmt.Errorf("failed to %s URL %s, got: %s", req.Method, req.URL, resp.Status)
	}
```
The best we can do without updating go-puppetca is checking the content of the error